dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/security/DefaultSecurityService.java

```diff
@@ -617,4 +627,27 @@ public boolean hasAnyAuthority( String... authorities )
 
         return false;
     }
+
+    @Override
+    public Map<String, Object> verifyRecaptcha( String key, String remoteIp )
+        throws IOException
+    {
+        MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+
+        params.add( "secret", (String) systemSettingManager.getSystemSetting( SettingKey.RECAPTCHA_SECRET ) );
+        params.add( "response", key );
+        params.add( "remoteip", remoteIp );
+
+        String result = restTemplate.postForObject( RECAPTCHA_VERIFY_URL, params, String.class );
+
+        log.info( "Recaptcha result: " + result );
+
+        ObjectMapper mapper = new ObjectMapper();
+        TypeReference<HashMap<String, Object>> typeRef
+            = new TypeReference<HashMap<String, Object>>()
+        {
+        };
+
```

Comment:
Any chance we can use

JacksonUtils.java
readValue

or make a new method in JacksonUtils to encapsulate the deserilization?